src/signer_node/message_processor/mod.rs

```diff
@@ -40,6 +56,141 @@ fn get_valid_block(state: &NodeState, blockhash: Hash) -> Result<&Block, Error>
     }
 }
 
+/// Create own VSSs and send to each other signers.
+/// Returns
+///     * own random key pair
+///     * a VSS for itself(for positive and negative)
+///     * own commitments
+pub fn create_block_vss<T, C>(
+    block: Block,
+    params: &NodeParameters<T>,
+    conman: &C,
+) -> (Keys, SharedSecret, SharedSecret)
+where
+    T: TapyrusApi,
+    C: ConnectionManager,
+{
+    let sharing_params = params.sharing_params();
+    let key = Sign::create_key(params.self_node_index + 1, None);
+
+    let parties = (0..sharing_params.share_count)
+        .map(|i| i + 1)
+        .collect::<Vec<usize>>();
+
+    let (vss_scheme, secret_shares) = VerifiableSS::share_at_indices(
+        sharing_params.threshold,
+        sharing_params.share_count,
+        &key.u_i,
+        &parties,
+    );
+    let order: BigInt = FE::q();
+    let (vss_scheme_for_negative, secret_shares_for_negative) = VerifiableSS::share_at_indices(
+        sharing_params.threshold,
+        sharing_params.share_count,
+        &(ECScalar::from(&(order - key.u_i.to_big_int()))),
+        &parties,
+    );
+
+    for i in 0..params.pubkey_list.len() {
+        // Skip broadcasting if it is vss for myself. Just return this.
+        if i == params.self_node_index {
+            continue;
+        }
+
+        conman.send_message(Message {
+            message_type: MessageType::BlockGenerationRoundMessages(
+                BlockGenerationRoundMessageType::Blockvss(
+                    block.sighash(),
+                    vss_scheme.clone(),
+                    secret_shares[i],
+                    vss_scheme_for_negative.clone(),
+                    secret_shares_for_negative[i],
+                ),
+            ),
+            sender_id: params.signer_id,
+            receiver_id: Some(SignerID {
+                pubkey: params.pubkey_list[i],
+            }),
+        });
+    }
+
+    (
+        key,
+        SharedSecret {
+            vss: vss_scheme.clone(),
+            secret_share: secret_shares[params.self_node_index],
+        },
+        SharedSecret {
+            vss: vss_scheme_for_negative.clone(),
+            secret_share: secret_shares_for_negative[params.self_node_index],
+        },
+    )
+}
+
+fn generate_local_sig<T>(
```

Comment:
this function should be move process_localsig.rs?

Reply:
No, it is also being used multiple place in message_processor module.